flink-libraries/flink-table/src/main/java/org/apache/flink/table/plan/nodes/datastream/DataStreamProcTimeTimeAggregate.java

```diff
@@ -0,0 +1,162 @@
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.plan.nodes.datastream;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.calcite.plan.RelOptCluster;
+import org.apache.calcite.plan.RelTraitSet;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rel.logical.LogicalWindow;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeField;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.java.typeutils.RowTypeInfo;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.functions.AssignerWithPeriodicWatermarks;
+import org.apache.flink.streaming.api.watermark.Watermark;
+import org.apache.flink.streaming.api.windowing.assigners.GlobalWindows;
+import org.apache.flink.streaming.api.windowing.evictors.TimeEvictor;
+import org.apache.flink.streaming.api.windowing.time.Time;
+import org.apache.flink.streaming.api.windowing.triggers.CountTrigger;
+import org.apache.flink.table.api.StreamTableEnvironment;
+import org.apache.flink.table.api.TableConfig;
+import org.apache.flink.table.calcite.FlinkTypeFactory;
+import org.apache.flink.table.plan.logical.rel.util.WindowAggregateUtil;
+import org.apache.flink.table.plan.nodes.datastream.DataStreamRel;
+import org.apache.flink.table.plan.nodes.datastream.function.DataStreamProcTimeAggregateGlobalWindowFunction;
+import org.apache.flink.table.plan.nodes.datastream.function.DataStreamProcTimeAggregateWindowFunction;
+import org.apache.flink.types.Row;
+
+public class DataStreamProcTimeTimeAggregate extends DataStreamRelJava {
+
+	private LogicalWindow windowReference;
+	private String description;
+
+	public DataStreamProcTimeTimeAggregate(RelOptCluster cluster, RelTraitSet traitSet, RelNode input,
+			RelDataType rowType, String description, LogicalWindow windowReference) {
+		super(cluster, traitSet, input);
+
+		this.rowType = rowType;
+		this.description = description;
+		this.windowReference = windowReference;
+
+	}
+
+	@Override
+	protected RelDataType deriveRowType() {
+		// TODO Auto-generated method stub
+		return super.deriveRowType();
+	}
+
+	@Override
+	public RelNode copy(RelTraitSet traitSet, java.util.List<RelNode> inputs) {
+
+		if (inputs.size() != 1) {
+			System.err.println(this.getClass().getName() + " : Input size must be one!");
+		}
+
+		return new DataStreamProcTimeTimeAggregate(getCluster(), traitSet, inputs.get(0), getRowType(),
+				getDescription(), windowReference);
+
+	}
+
+	@Override
+	public DataStream<Row> translateToPlan(StreamTableEnvironment tableEnv) {
+
+		// Get the general parameters related to the datastream, inputs, result
+		TableConfig config = tableEnv.getConfig();
+
+		DataStream<Row> inputDataStream = ((DataStreamRel) getInput()).translateToPlan(tableEnv);
+
+		TypeInformation<?>[] rowType = new TypeInformation<?>[getRowType().getFieldList().size()];
```

Comment:
Not sure how you suggest to make the checks? Is there a style in flink for doing this?

Also related to the suggestion of using a variable for getRowType().getFieldList() - I will do it, but in theory modern day compilers such as JAVA should identify and optimize this kind of things. And therefore I would expect the result is the same.